src/module/apps/compendium-browser/index.ts

```diff
@@ -331,14 +466,70 @@ class CompendiumBrowser extends Application {
                         }
                     }
                     await game.settings.set("pf2e", "compendiumBrowserPacks", this.settings);
-                    for (const tab of Object.values(this.tabs)) {
-                        if (tab.isInitialized) {
-                            await tab.init();
-                            tab.scrollLimit = 100;
+
+                    for (const [key, source] of Object.entries(this.packLoader.sources.allowed)) {
+                        if (!source || isBlank(source.name)) {
+                            delete this.packLoader.sources.allowed[key]; // just to make sure we clean up
+                            continue;
                         }
+                        source.load = formData.has(`source-${key}`);
                     }
+
+                    this.packLoader.sources.allowEmptySources = formData.has("allow-empty-sources");
+                    this.packLoader.sources.allowUnknownSources = formData.has("allow-unknown-sources");
+                    await game.settings.set("pf2e", "compendiumBrowserSources", this.packLoader.sources);
+
+                    await this.#resetInitializedTabs();
                     this.render(true);
                 });
+
+                const sourceSearch = htmlQuery<HTMLInputElement>(form, "input[data-element=setting-sources-search]");
+                const sourceToggle = htmlQuery<HTMLInputElement>(
+                    form,
+                    "input[data-action=setting-sources-toggle-visible]"
+                );
+                const sourceSettings = htmlQueryAll<HTMLElement>(form, "label[data-element=setting-source]");
+
+                sourceSearch?.addEventListener("input", () => {
+                    const value = sourceSearch.value?.trim().toLocaleLowerCase(game.i18n.lang);
+
+                    for (const element of sourceSettings) {
+                        const shouldBeHidden =
+                            !isBlank(value) &&
+                            !isBlank(element.dataset.name) &&
+                            element.dataset.name.toLocaleLowerCase(game.i18n.lang).indexOf(value) === -1;
```

Comment:
```suggestion
                    for (const element of sourceSettings) {
                        const name = element.dataset.name?.toLocaleLowerCase(game.i18n.lang);
                        const shouldBeHidden = !isBlank(value) && !isBlank(name) && !name.includes(value);
```
Other than that it looks good to me now.

Reply:
This is updated.